src/lib.rs

```diff
@@ -402,10 +399,62 @@ struct NodeData<'input> {
     pos: usize,
 }
 
+/// A string storage.
+///
+/// Used by text nodes and attributes values.
+///
+/// We try our best to keep parsed strings as `&str`, but in some cases post-processing
+/// is necessary and we have to allocates them.
+///
+/// All owned, allocated strings are stored as `Arc<str>`.
+#[derive(Clone, Eq, Debug)]
+pub enum StringStorage<'input> {
+    /// A raw slice of the input string.
+    Borrowed(&'input str),
+
+    /// A reference-counted string.
+    Owned(alloc::sync::Arc<str>),
+}
+
+impl StringStorage<'_> {
+    /// Creates a new owned string from `&str` or `String`.
+    pub fn new_owned<T: Into<alloc::sync::Arc<str>>>(s: T) -> Self {
```

Comment:
Does this method really carry its weight now? The enum variant is public in any case and the trait bounds are the same as when constructing the enum at the single place where this is used?

Reply:
The difference is that when I have to create my own `StringStorage`, which I do, it will be:

```rust
StringStorage::new_owned("text")
// instead of
StringStorage::Owned(Arc::from("text"))
```

A nice shorthand.